File: py/ege9.170v2.py
f = open("ege9.170.txt")
gax = (list(map(int, l.split())) for l in f if l != '\n')

n = 0

for ax in gax:
    OneRep = False
    repCounts = []
    noreps = []

    for x in ax:
        repCounts.append(ax.count(x))

    if repCounts.count(1) != 4: continue

    for x in range(len(ax)):
        if repCounts[x] == 1: noreps.append(ax[x])

    Srep = sum(ax) - sum(noreps)

    if sum(noreps) <= Srep * 4:
        n += 1

print(n)

File: c/ege9.170v2.c
#include <stdio.h>
#include "array.lib.c"

#define N 6
#define R1 4

int main() {
    FILE *f = fopen("ege9.170.txt", "r");
    int a[N];
    int b[N];
    int c[N];
    int n = 0;

    while(freadn(f, a, N) == N) {
        for(int i = 0; i < N; i++) {
            b[i] = count(a, N, a[i]);
            c[i] = (b[i] == 1) ? a[i] : 0;
        }

        if(count(b, N, 1) != R1) continue;

        int sNoRep = sum(c, N);
        int sRep = sum(a, N) - sNoRep;

        if(sNoRep <= sRep * 4) n++;
        
    }
    printf("%d\n", n);
}
